e2e/nomostest/artifactregistry/image.go

```diff
@@ -64,7 +63,7 @@ func SetupImage(nt *nomostest.NT, name, version string) (*Image, error) {
 		// Use test name to avoid overlap. Truncate to 40 characters.
 		Name: generateImageName(nt, name),
 		// Use random suffix to avoid overlap. Truncate to 20 characters.
```

Comment:
We either need to truncate to 20 characters or just error if the name is too long. Error might be better if it needs to be semver for helm. And this comment need updating.

Reply:
version length check has been added and will error out when it's longer than 20 chars. Comments have been updated